src/charm.py

```diff
@@ -378,6 +380,17 @@ def _post_snap_refresh(self, refresh: charm_refresh.Machines):
 
         Called after snap refresh
         """
+        try:
+            if raw_cert := self.get_secret(UNIT_SCOPE, "internal-cert"):
+                cert = load_pem_x509_certificate(raw_cert.encode())
+                if (
+                    cert.subject.get_attributes_for_oid(NameOID.COMMON_NAME)[0].value
+                    != self._unit_ip
+                ):
+                    self.tls.generate_internal_peer_cert()
```

Comment:
Please add INFO-like message to log to see when cert is regenerated (to simplify production troubleshooting). Warning is also fine.